src/main/java/io/cdap/plugin/http/sink/batch/HTTPRecordWriter.java

```diff
@@ -109,13 +108,11 @@ private void executeHTTPService() throws IOException {
     do {
       HttpURLConnection conn = null;
 
-      Map<String, String> headers = null;
+      Map<String, String> headers = config.getRequestHeadersMap();
+
+      Header authorizationHeaders = config.getAuthorizationHeaders();
+      headers.putAll(config.getHeadersMap(String.valueOf(authorizationHeaders)));
```

Comment:
Although it does not result into NullPointerException here, better to add a null check for authorizationHeader